src/utils/helpers.ts

```diff
@@ -420,3 +420,21 @@ export function parseDate(date: string | number | Date): Date {
 
   return givenDate;
 }
+
+/**
+ * Returns an array of all possible combinations consisting of one member
+ * from each of passed arrays
+ * @param arrays - set of arrays
+ */
+export function cartesian<T extends unknown[][]>(
+  ...arrays: T
+): MapCartesian<T>[] {
+  return arrays.reduce(
+    (a, b) => a.flatMap((d) => b.map((e) => [d, e].flat())),
+    [[]]
+  ) as MapCartesian<T>[];
+}
+
+type MapCartesian<T extends unknown[][]> = {
+  [K in keyof T]: T[K] extends Array<infer U> ? U : never;
+};
```

Comment:
I tried to type this to make it easier to work with - idea is from [here](https://stackoverflow.com/questions/65025411/how-to-utilise-typescript-variadic-tuple-types-for-a-cartesian-product-function)

Reply:
This is really cool!